src/RetargetingHelper/src/Helper.cpp

```diff
@@ -26,112 +26,328 @@ void RetargetingClient::convertYarpVectorPoseIntoTransform(const yarp::sig::Vect
+                    okCoMHeight = true;
+                }
+            }
+
+            if(okCoMHeight)
+                return true;
+
+            yarp::os::Time::delay(0.001);
+            attempt++;
+        } while (attempt < 100);
+
+        if(!okCoMHeight)
+            yError() << "[RetargetingClient::reset] The CoM height is not coming from the yarp port.";
+        return false;
+    }
 
-    m_leftHandSmoother->init(m_leftHandTransformYarp);
-    m_rightHandSmoother->init(m_rightHandTransformYarp);
+    return true;
 }
 
 void RetargetingClient::getFeedback()
 {
-    if(!m_useHandRetargeting)
-        return;
+    if(m_useHandRetargeting)
+    {
+        auto desiredLeftHandPose = m_leftHand.port.read(false);
+        if(desiredLeftHandPose != nullptr)
+            m_leftHand.yarpVector = *desiredLeftHandPose;
 
-    auto desiredLeftHandPose = m_leftHandTransformPort.read(false);
-    if(desiredLeftHandPose != nullptr)
-        m_leftHandTransformYarp = *desiredLeftHandPose;
+        m_leftHand.smoother->computeNextValues(m_leftHand.yarpVector);
+        convertYarpVectorPoseIntoTransform(m_leftHand.smoother->getPos(), m_leftHandTransform);
 
-    m_leftHandSmoother->computeNextValues(m_leftHandTransformYarp);
-    convertYarpVectorPoseIntoTransform(m_leftHandSmoother->getPos(), m_leftHandTransform);
+        auto desiredRightHandPose = m_rightHand.port.read(false);
+        if(desiredRightHandPose != nullptr)
+            m_rightHand.yarpVector = *desiredRightHandPose;
 
-    auto desiredRightHandPose = m_rightHandTransformPort.read(false);
-    if(desiredRightHandPose != nullptr)
-        m_rightHandTransformYarp = *desiredRightHandPose;
+        m_rightHand.smoother->computeNextValues(m_rightHand.yarpVector);
+        convertYarpVectorPoseIntoTransform(m_rightHand.smoother->getPos(), m_rightHandTransform);
 
-    m_rightHandSmoother->computeNextValues(m_rightHandTransformYarp);
-    convertYarpVectorPoseIntoTransform(m_rightHandSmoother->getPos(), m_rightHandTransform);
+
+        yInfo() << "ciao hand";
+    }
+
+    if(m_useJointRetargeting)
+    {
+        auto desiredJoint = m_jointRetargeting.port.read(false);
+        if(desiredJoint != nullptr)
+        {
+            for(int i =0; i < desiredJoint->size(); i++)
+                m_jointRetargeting.yarpVector(m_retargetJointsIndex[i]) = (*desiredJoint)(i);
+        }
+
+        m_jointRetargeting.smoother->computeNextValues(m_jointRetargeting.yarpVector);
+        iDynTree::toEigen(m_jointRetargetingValue) = iDynTree::toEigen(m_jointRetargeting.smoother->getPos());
+    }
+
+
+    if(m_useCoMHeightRetargeting)
+    {
+        if(!m_isStancePhase)
+            m_comHeight.yarpVector(0) = m_comConstantHeight;
+        else
+        {
+            auto desiredCoMHeight = m_comHeight.port.read(false);
+            if(desiredCoMHeight != nullptr)
+
+                m_comHeight.yarpVector(0) = ((*desiredCoMHeight)(2) - m_comHeightInputZero) * 0.5
```

Comment:
Is the 0.5 a configuration parameter?

Reply:
Magic numbers :cry: :

Reply:
I remember that this heuristic was added as a workaround for an Xsens problem. As far as I understood the IK used to compute the `desiredCoMHeight` uses the model of the robot and the base directly coming from the Xsens. This means that the observer implemented in the Xsens generates a base displacement according to the human model. 
In other words, the CoM position is computed as follows:

![Untitled Diagram (7)](https://user-images.githubusercontent.com/16744101/80352181-c7079180-8873-11ea-87a5-16ec44fbbfbf.png)

The desired position of the CoM should depend on the desired position of the robot base (i.e. the human base pose scaled taking into account the difference between the robot and the human size)

The problem can be seen [here](https://www.youtube.com/watch?v=yELyMYkCyNE&feature=youtu.be&t=26). 

On the light of this we should decide, or we:
1. remove the CoM retargeting
2. fix the problem in the Xsens side

@S-Dafarra and @kouroshD

Reply:
I see. I think the problem here is that it is not completely clear which kind of information the walking controller is expecting. If the controller is expecting the height given the inertial frame it is considering, then the retargeting needs to know how this frame is located. In other words, the retargeting block needs to receive the base pose from the controller. I would argue that this is not the best choice since a reference quantity would depend on a measured quantity. 
The real question then is: which kind of reference the retargeting should generate?
Bonus question: how can we define this reference such that we can seamlessly switch from the XSens to the virtualizer?

Anyways, for the sake of this PR, I would remove it entirely, open an issue, and work on a new PR in order to add this feature, especially if this needs to be paired with a PR on ``walking-teleoperation``. I would also stress that the CoM retargeting was preventing the problem when raising the arms, so we should not let this fall in the oblivion.

Reply:
I agree so I will remove the CoM retargeting from the code

Reply:
Otherwise, I can keep the code as it is and disabling the CoM retargeting setting `m_useCoMHeightRetargeting = false`. This may help us in the future when we will reimplement the feature

Reply:
Ok, but either throw a warning, add a comment in the configuration, or whatever you prefer just to remind us of the status. 
If possible, I would move also all the magic numbers in the configuration files, but this would not be blocking.
Anyways, it is better to open an issue and start discussing on how to face this.

Reply:
The idea was to remove everything from the configuration file and disable the `CoMHeightRetargeting` adding 
```cpp
const m_useCoMHeightRetargeting = false
```
as an attribute of the class. If you don't like it I can remove everything